### PR TITLE
Allow EX channel role change

### PR DESCRIPTION
**Pull request by norkans7**
Fixes https://github.com/rapidpro/rapidpro/issues/577

### CODE REVIEW

**Comment by norkans7** [comment]
includes #3037

**Comment by rowanseymour** [comment]
@norkans7 sorry think I caused some conflicts here.. also is supposed to be included again in https://github.com/nyaruka/rapidpro/pull/3037 ? That diff has the same changes

**Comment by rowanseymour** [comment]
@norkans7 sorry to clarify, you're saying this PR includes https://github.com/nyaruka/rapidpro/pull/3037 but it looks like https://github.com/nyaruka/rapidpro/pull/3037 includes this PR no?

**Comment by norkans7** [comment]
#3037 needs to be merged first, I had worked from the same branch

The changes here are only in https://github.com/nyaruka/rapidpro/pull/3038/commits/783513fa6936d9b57b26f085ecf7af75e8b08a53

**Comment by rowanseymour** [comment]
Ah ok - I saw the PR title of "Fix export styles" and the diff that is mostly changes to channel types and thought maybe it included something it shouldn't

File: temba/channels/types/external/views.py [hunk 1]
```diff
@@ -223,3 +223,20 @@ def form_valid(self, form):
...
+        help_text=_("The roles this channel can fulfill"),
+    )
+
+    def clean_role(self):
+        return "".join(self.cleaned_data.get("role", []))
+
+    class Meta(UpdateTelChannelForm.Meta):
+        fields = "name", "alert_email", "role"
+        readonly = []
+        helps = {"address": _("Phone number of this device")}
```
Comment (rowanseymour) [inline temba/channels/types/external/views.py:242]: address isn't a field... no?

**Review by rowanseymour — CHANGES_REQUESTED** [review]

**Comment by rowanseymour** [comment]
@norkans7 seems this would require a change in goflow https://github.com/nyaruka/goflow/issues/974

**Comment by norkans7** [comment]
Added https://github.com/nyaruka/courier/pull/338 and https://github.com/nyaruka/goflow/pull/975

**Pull request merged**